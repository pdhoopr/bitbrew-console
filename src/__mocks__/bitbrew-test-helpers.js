import { shallow } from 'enzyme';
import enzymeSerializer from 'enzyme-to-json/serializer';
import diff from 'jest-diff';
import { addSerializer, getSerializers, utils } from 'jest-snapshot';
import styledSerializer from 'jest-styled-components/src/styleSheetSerializer';
import stripAnsi from 'strip-ansi';

const DIFF_HEADER = 'Diff:\n';

export const diffComponents = (baseComponent, compareComponent) => {
  const serializers = getSerializers();
  [styledSerializer, enzymeSerializer].forEach(serializer => {
    if (!serializers.includes(serializer)) {
      addSerializer(serializer);
    }
  });
  const base = `${utils.serialize(shallow(baseComponent))}\n`;
  const compare = `${utils.serialize(shallow(compareComponent))}\n`;
  const changes = diff(base, compare, {
    aAnnotation: `<BaseComponent />`,
    bAnnotation: `<CompareComponent />`,
    contextLines: 3,
    expand: false,
  });
  return `${DIFF_HEADER}${changes}`;
};

export const diffSerializer = {
  test(value) {
    return typeof value === 'string' && value.startsWith(DIFF_HEADER);
  },
  print(value) {
    return stripAnsi(value).trim();
  },
};

export const fixtures = {
  accessToken:
    'eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJPbmxpbmUgSldUIEJ1aWxkZXIiLCJpYXQiOjE1MjUxMTUzNTAsImV4cCI6MTU1NjY1MTM1MCwiYXVkIjoid3d3LmV4YW1wbGUuY29tIiwic3ViIjoianJvY2tldEBleGFtcGxlLmNvbSJ9.4O-W8aeOOUFU8fkGTAFaAu9D9nK_l61yPomNG1Qkny8',
  orgs: {
    danlaw: {
      id: 'bd006616-3bd2-4783-a84d-b0fa4ed81135',
      properName: 'Danlaw',
    },
    azuga: {
      id: '8d9ec637-f3bb-4c46-83ee-11764d09f0b2',
      properName: 'Azuga',
    },
    bitbrew: {
      id: '94d9b0ca-a758-4502-8dcd-41d883e38714',
      properName: 'BitBrew',
    },
  },
  users: {
    bill: {
      email: 'bill@bitbrew.com',
      name: 'Bill Monaghan',
    },
    cynthia: {
      email: 'cynthia@bitbrew.com',
      name: 'Cynthia Hamilton',
    },
    alex: {
      email: 'bill@bitbrew.com',
      name: 'Alex Li',
    },
  },
};
