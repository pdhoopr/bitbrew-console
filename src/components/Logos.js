import styled, { css } from 'styled-components';
import Mark from '../graphics/logomark.svg';
import Type from '../graphics/logotype.svg';

export const baseStyles = css`
  display: block;
  height: var(--size-40);
  width: auto;
`;

export const Logomark = styled(Mark)`
  ${baseStyles};
`;

export const Logotype = styled(Type)`
  ${baseStyles};
`;
