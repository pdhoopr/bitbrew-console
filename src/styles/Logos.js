import styled, { css } from 'styled-components';
import Mark from '../graphics/logomark.svg';
import Type from '../graphics/logotype.svg';

const baseStyles = css`
  display: block;
  height: var(--size-40);
  width: auto;
`;

export const Logomark = styled(Mark)`
  ${baseStyles};
`;

export const Logotype = styled(Type)`
  ${baseStyles};
`;
