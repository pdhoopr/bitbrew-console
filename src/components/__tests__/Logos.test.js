import { diffComponents } from 'bitbrew-test-helpers';
import { shallow } from 'enzyme';
import React from 'react';
import styled from 'styled-components';
import { baseStyles, Logomark, Logotype } from '../Logos';

const BaseLogo = styled.svg`
  ${baseStyles};
`;

test('renders DOM shared by all logos', () => {
  const dom = shallow(<BaseLogo />);
  expect(dom).toMatchSnapshot();
});

test('renders DOM specific to the logomark', () => {
  const base = <BaseLogo />;
  const comp = <Logomark />;
  const diff = diffComponents(base, comp);
  expect(diff).toMatchSnapshot();
});

test('renders DOM specific to the logotype', () => {
  const base = <BaseLogo />;
  const comp = <Logotype />;
  const diff = diffComponents(base, comp);
  expect(diff).toMatchSnapshot();
});
