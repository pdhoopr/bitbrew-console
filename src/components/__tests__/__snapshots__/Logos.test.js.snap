// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders DOM shared by all logos 1`] = `
.c0 {
  display: block;
  height: var(--size-40);
  width: auto;
}

<svg
  className="c0"
/>
`;

exports[`renders DOM specific to the logomark 1`] = `
Diff:
Compared values have no visual difference.
`;

exports[`renders DOM specific to the logotype 1`] = `
Diff:
Compared values have no visual difference.
`;
